rascal-vscode-extension/package.json

```diff
@@ -138,6 +138,27 @@
         "path": "./syntaxes/parametric.tmGrammar.json"
       }
     ],
+    "semanticTokenScopes": [
```

Comment:
~Can you summarize what this section does?~

---

Actually, I decided to ask GitHub Copilot Chat about this snippet of code with no other context. Is it right?

> In a vs code extension, what does this block mean?

<details><summary>The answer</summary>
<p>
This JSON block is part of a Visual Studio Code extension's configuration, specifically related to semantic highlighting.

Semantic highlighting is a feature that provides more granular syntax coloring based on the semantic context, rather than just the type of the token. This can make it easier to understand the structure and meaning of your code.

Here's a breakdown of the JSON:

- `"semanticTokenScopes"`: This is an array of objects, each defining the semantic token scopes for a specific language. In this case, two languages are defined: `rascalmpl` and `parametric-rascalmpl`.

  - `"language"`: The language identifier.
  - `"scopes"`: An object mapping semantic token types to textmate scopes. In this case, the `ambiguity` token type is mapped to the `invalid` textmate scope.

- `"semanticTokenTypes"`: This is an array of objects, each defining a semantic token type.

  - `"id"`: The identifier of the semantic token type.
  - `"superType"`: The super type of this semantic token type. This is used for inheritance of properties.
  - `"description"`: A human-readable description of the semantic token type.

In this case, the `ambiguity` token type is defined, which is used to highlight ambiguous parts of a parse tree in the Rascal language. The `invalid` scope is typically used to highlight errors or problems in code, so it seems this extension uses it to highlight these ambiguous parts.
</p>
</details>

Reply:
It's documented in these two sections: 
- https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#custom-token-types-and-modifiers 
- https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#custom-textmate-scope-mappings